### PR TITLE
Reach the max level for phpstan

### LINKED ISSUE #46: Static Analysis level
Both tools should be at their maximum level, with a baseline. Let's work our way up, fixing obvious issues, until we give up and baseline the rest.

### PR DESCRIPTION
**Pull request by stof**
Closes #46

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

File: phpstan.neon
Comment (greg0ire) [inline phpstan.neon:2]: Should we use the `max` alias here?

**Comment by greg0ire** [comment]
Also, now that #58 is merged, the description can be changed to "Closes #46" I think.

**Comment by derrabus** [comment]
This PR needs a rebase.

**Comment by stof** [comment]
PR is rebased

File: phpstan.neon
Comment (greg0ire) [reply phpstan.neon:2]: ping @stof

Comment (stof) [reply phpstan.neon:2]: the question is whether we would want to automatically bump to level 10 in case phpstan introduces a new higher level (even though I don't see what it would container to be stricter than level 9).
phpstan did that in the past. For a long time, there was no level 9.

Comment (greg0ire) [reply phpstan.neon:2]: I think so yes, because I don't think we are going to remember to bump it to level 10, and also because since that library is small, it should not result in a lot of errors. Even if it did, we could still baseline them and hold contributions to a higher standard immediately, instead of letting in more code that needs to be fixed by maintainers.

Comment (derrabus) [reply phpstan.neon:2]: I'm fine with either setting. 🤷🏻‍♂️

**Review by derrabus — APPROVED** [review]

File: phpstan.neon
Comment (greg0ire) [reply phpstan.neon:2]: Me too actually, so let's not block this for days. When there is a new level, I think we will be aware and can bump this on all repositories.

**Pull request merged**
